gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2227,71 +2271,30 @@ void Movable::FixPosition()
 	Pos.y=Pos.y*12+6;
 }
 
-void Movable::WalkTo(const Point &Des, int distance)
-{
-	Point from;
-
-	// maybe caller should be responsible for this
-	if ((Des.x/16 == Pos.x/16) && (Des.y/12 == Pos.y/12)) {
-		ClearPath();
-		return;
-	}
+void Movable::WalkTo(const Point &Des, int distance) {
+	Log(DEBUG, "WalkToWIP", "curTime = %d, prevTicks = %d", Ticks, this->prevTicks);
+	// The rate limiting stuff is probably not optimal, since it requires lots of pathfinder calls
```

Comment:
Did you look into the sources of WalkTo calls? I would generally expect it to be called once per tick, not more.